gateway/api_loader.go

```diff
@@ -997,6 +998,10 @@ func (gw *Gateway) loadApps(specs []*APISpec) {
 
 	gw.apisMu.Unlock()
 
+	for _, spec := range specsToRelease {
```

Comment:
Move up in front of .Unlock?

Reply:
`apisMu` is for `gw.apisByID` and `gw.apisHandlesByID`. Here we're collecting `*APISpec` to be released. I think it's okay to leave it outside the mutex. Also it won't keep the mutex in lock state for long since `spec.Release()` is called sequentially.